app/src/main/java/com/garethevans/church/opensongtablet/NearbyConnections.java

```diff
@@ -249,12 +253,16 @@ public void onDisconnected(@NonNull String endpointId) {
                 String deviceName = getDeviceNameFromId(endpointId);
                 connectedEndPointsNames.remove(endpointId+"__"+deviceName);
                 updateConnectionLog(context.getResources().getString(R.string.connections_disconnect) + " " + deviceName);
-                // Check if we have valid connections
-                StaticVariables.isConnected = stillValidConnections();
-                // Try to connect again after 2 seconds
-                if (!StaticVariables.isHost) {
-                    Handler h = new Handler();
-                    h.postDelayed(() -> startDiscovery(), 2000);
+                if (StaticVariables.isHost) {
+                    // Check if we have valid connections
+                    StaticVariables.isConnected = stillValidConnections();
+                } else {
+                    // Try to connect again after 2 seconds
+                    if (!StaticVariables.isHost) {
```

Comment:
Sometime you do not spot the obvious - please remove the unneeded 'if (!StaticVariables.isHost)'.